chromium_src/chrome/browser/component_updater/widevine_cdm_component_installer.cc

```diff
@@ -3,39 +3,329 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include "chrome/browser/component_updater/widevine_cdm_component_installer.h"
+
+#if BUILDFLAG(WIDEVINE_ARM64_DLL_FIX)
+#define WidevineCdmComponentInstallerPolicy \
+  WidevineCdmComponentInstallerPolicy_ChromiumImpl
+#endif
+
 #define RegisterWidevineCdmComponent RegisterWidevineCdmComponent_ChromiumImpl
 #include "src/chrome/browser/component_updater/widevine_cdm_component_installer.cc"
 #undef RegisterWidevineCdmComponent
 
+#if BUILDFLAG(WIDEVINE_ARM64_DLL_FIX)
+#undef WidevineCdmComponentInstallerPolicy
+#endif
+
+#include "base/check.h"
+#include "base/functional/callback.h"
 #include "brave/browser/widevine/widevine_utils.h"
-#include "chrome/browser/component_updater/component_updater_utils.h"
+#include "brave/components/widevine/static_buildflags.h"
 #include "components/component_updater/component_updater_service.h"
-#include "extensions/common/constants.h"
 
-namespace component_updater {
+#if BUILDFLAG(WIDEVINE_ARM64_DLL_FIX)
 
-namespace {
+#include "base/feature_list.h"
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/json/json_reader.h"
+#include "base/json/json_writer.h"
+#include "base/logging.h"
+#include "base/metrics/field_trial_params.h"
+#include "base/synchronization/waitable_event.h"
+#include "base/threading/thread_restrictions.h"
+#include "base/time/time.h"
+#include "base/values.h"
+#include "components/update_client/task_traits.h"
+#include "components/update_client/utils.h"
+#include "net/base/net_errors.h"
+#include "net/traffic_annotation/network_traffic_annotation.h"
+#include "services/network/public/cpp/resource_request.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "third_party/zlib/google/zip.h"
+#include "url/gurl.h"
```

Comment:
these headers should be included w/o `WIDEVINE_ARM64_DLL_FIX`. need to sync with gn configration.
If above header files are not guarded in gn, we don't need to use another guard in source code.

Reply:
@simonhong adding the headers didn't make any additional GN dependencies necessary. Having the guard makes it easy to remove the Widevine workaround once upstream officially offers native Arm64 binaries. Can we keep the guard?

Reply:
Ah, ok for that purpose :)